kolibri/core/public/utils.py

```diff
@@ -222,8 +223,9 @@ def request_soud_sync(server, user, queue_id=None, ttl=4):
             # got a place in the queue to sync with this server, so we can be
             # more sure that the server is actually available.
             response = requests.put(server_url, json=data, timeout=30)
-
-    except requests.exceptions.ConnectionError:
+        if response.status_code >= status.HTTP_500_INTERNAL_SERVER_ERROR:
+            raise ConnectionError()
```

Comment:
I'm assuming there's a reason we don't want to use `raise_for_status` here, yeah? Are there cases where we want 4xx response statuses to be swallowed?

Reply:
Yeah - any 4xx error codes indicate that the server has received and processed the request, and is not interested, so we just stop trying.